osquery/tables/system/windows/users.cpp

```diff
@@ -197,9 +246,24 @@ void processLocalAccounts(std::set<std::string>& processedSids,
 QueryData genUsers(QueryContext& context) {
   QueryData results;
   std::set<std::string> processedSids;
+  std::set<std::string> selectedUids;
 
-  processLocalAccounts(processedSids, results);
-  processRoamingProfiles(processedSids, results);
+  // implement index on UUID (SID on Windows) column by returning only the
+  // users in the constraint, bypassing the enumeration step entirely:
+  if (context.constraints["uuid"].exists(EQUALS)) {
+    auto sidStrings = context.constraints["uuid"].getAll(EQUALS);
+    for (const auto& sidString : sidStrings) {
+      genUser(sidString, results);
+    }
+  } else {
```

Comment:
Prefer early return over if-else

Reply:
Actually I think this `if`/`else` setup is fine? If one did an early `return` here you'd have two `return` statements. Left as is we always update `results` and have the single return at the end. Thoughts?

Reply:
Hrm. I'm generally okay with multiple returns. At least, I find it cleaner than parsing nested if blocks.

But I also don't think we should block on style here

Reply:
If I may be the tiebreaker opinion then, I'd like to leave it this way, and it's a short routine anyway.